geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java

```diff
@@ -89,12 +93,51 @@ public Object execute(Object command, ParseResult parseResult) {
     }
   }
 
-  protected Object invokeCommand(Object command, ParseResult parseResult) {
+  protected Object invokeCommand(Object command, GfshParseResult parseResult) {
     // if no command instance is passed in, use the one in the parseResult
     if (command == null) {
       command = parseResult.getInstance();
     }
-    return ReflectionUtils.invokeMethod(parseResult.getMethod(), command,
-        parseResult.getArguments());
+
+    Object result =
+        ReflectionUtils.invokeMethod(parseResult.getMethod(), command, parseResult.getArguments());
+
+    if (!(command instanceof SingleGfshCommand)) {
+      return result;
+    }
+
+    SingleGfshCommand gfshCommand = (SingleGfshCommand) command;
+    CommandResult commandResult = (CommandResult) result;
+    if (commandResult.getStatus() == Result.Status.ERROR) {
+      return result;
+    }
+
+    // if command result is ok, we will need to see if we need to update cluster configuration
+    ClusterConfigurationService ccService = gfshCommand.getConfigurationService();
+    if (parseResult.getParamValue("member") != null || ccService == null) {
+      commandResult.setCommandPersisted(false);
+      return commandResult;
+    }
+
+    String groupInput = parseResult.getParamValueAsString("group");
+    if (groupInput == null) {
+      groupInput = "cluster";
+    }
+    String[] groups = groupInput.split(",");
+    for (String group : groups) {
+      ccService.updateCacheConfig(group, cc -> {
+        try {
+          gfshCommand.updateClusterConfig(group, cc, commandResult.getCacheElemnt());
```

Comment:
The `try` block probably shouldn't be in the mutator itself, should it..?  In conjunction with the above comment, if the command returned a mutator, then this block would instead read:

```
try{
  ccService.updateCacheConfig(group, gfshCommand.getConfigurationUpdater());
} catch (Exception e) {
  commandResult setCommandPersisted(false);
  commandResult.addLine(
    "cluster configuration for '" + group + "' is not updated. " + e.getMessage());
}
return commandResult;
```

Reply:
I immediately realize that the `try` block is for the stability of the `UnaryOperator`, which must only throw `RuntimeException`s.